haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,755 @@
+                    ]
+                    sub_units = hard_splits, [0] * len(hard_splits)
+                    sub_documents = self.split_into_units(
+                        document=document, units=sub_units, add_page_number=add_page_number
+                    )[0]
+                    if add_page_number:
+                        for sub_document in sub_documents:
+                            sub_document.meta["page"] += document.meta["page"] - 1
+                    sane_documents += sub_documents
+
+            final_documents += sane_documents
+        return {"documents": final_documents}, "output_1"
+
+    def run_batch(  # type: ignore
+        self,
+        documents: List[List[Document]],
+        split_by: Optional[SplitBy] = None,
+        split_regex: Optional[str] = None,
+        split_length: Optional[int] = None,
+        split_overlap: Optional[int] = None,
+        split_max_chars: Optional[int] = None,
+        add_page_number: Optional[bool] = None,
+    ):
+        """
+        Splits documents into smaller, shorter documents.
+
+        Can split on different units ('character', 'token', 'word', 'sentence', 'paragraph', 'page', or 'regex'),
+        at different lengths, and include some overlap across the splits.
+
+        It can also properly assign page numbers and re-assign headlines found in the metadata of the parent document
+        to each split document.
+
+        No char is lost in splitting, not even whitespace, and all headlines are preserved, However, text and headlines
+        duplication may occur if `split_overlap>0`.
+
+        :param documents: The documents to split.
+        :param split_by: Unit for splitting the document. Can be 'character', 'token', 'word', 'sentence', 'paragraph', 'page', 'regex'.
+        :param split_regex: If `split_by="regex"`, provide here a regex matching the separator. For example, if the document
+                            should be split on "--my separator--", this field should be `split_regex="--my separator--"`.
+        :param split_length: The maximum number of the above split unit (like word, sentence, page and so on) that are allowed in one document.
+                                For instance, if `split_lenght=10` and `split_by="sentence"`, then each output document will contain 10 sentences.
+        :param split_overlap: Units (for example words or sentences) overlap between two adjacent documents after a split.
+                                For example, if `split_by="word" and split_length=5 and split_overlap=2`, then the splits would be like:
+                                `[w1 w2 w3 w4 w5, w4 w5 w6 w7 w8, w7 w8 w10 w11 w12]`.
+                                Set the value to 0 to ensure there is no overlap among the documents after splitting.
+        :param max_chars: Absolute maximum number of chars allowed in a single document. Reaching this boundary
```

Comment:
I think `max_chars` is meant to be `split_max_chars` here based on the variable name in the `__init__` function.